src/plugins/intel_cpu/src/nodes/reverse_sequence.cpp

```diff
@@ -99,88 +80,109 @@ void ReverseSequence::initSupportedPrimitiveDescriptors() {
+}
 
-            parallel_nt(0, [&](const int ithr, const int nthr) {
-                size_t i, start = 0, end = 0, src_idx = 0;
-                SizeVector counters(src_dims.size(), 0);
-                splitter(work_amount_dst, nthr, ithr, start, end);
-                for (int j = src_dims.size() - 1, i = start; j >= 0; j--) {
-                    counters[j] = i % src_dims[j];
-                    i /= src_dims[j];
-                }
+void ReverseSequence::executeDynamicImpl(dnnl::stream strm) {
+    execute(std::move(strm));
+}
 
-                for (size_t iwork = start; iwork < end; ++iwork) {
-                    for (i = 0, src_idx = 0; i < src_dims.size(); ++i) {
-                        size_t idx = counters[i];
-                        if (static_cast<int>(i) == seq_axis &&
-                            static_cast<int>(idx) < static_cast<int32_t>(seq_lengths_data[counters[batch_axis]])) {
-                            idx = static_cast<int32_t>(seq_lengths_data[counters[batch_axis]]) - idx - 1;
-                        }
-                        src_idx += idx * srcStrides[i];
-                    }
-                    dst_data[iwork] = src_data[src_idx];
-                    for (int j = src_dims.size() - 1; j >= 0; j--) {
-                        counters[j] = (counters[j] + 1) % src_dims[j];
-                        if (counters[j] != 0) break;
-                    }
-                }
-            });
+ReverseSequence::ReverseSequenceExecutor::ReverseSequenceExecutor(const VectorDims& dataDims,
+    const VectorDims& seqLengthsDims, const VectorDims& dstDims, int batchAxis, int seqAxis)
+        : batchAxis{batchAxis}
+        , seqAxis{seqAxis} {
+    for (size_t i = 0; i < dataDims.size(); ++i) {
+        if (dataDims[i] != dstDims[i])
+            IE_THROW() << "Input/output tensors dimensions mismatch";
+    }
+
+    if (seqLengthsDims[0] != dataDims[batchAxis])
+        IE_THROW() << "'seq_lengths' dimension mismatch";
+
+    srcStrides.resize(dataDims.size());
+    srcStrides[srcStrides.size() - 1] = 1;
+    for (int i = srcStrides.size() - 2; i >= 0; --i) {
+        srcStrides[i] = srcStrides[i + 1] * dataDims[i + 1];
+    }
+
+    workAmountDst = srcStrides[0] * dataDims[0];
+}
+
+template<typename T>
+void ReverseSequence::ReverseSequenceExecutor::exec(const MemoryPtr& dataMemPtr, const MemoryPtr& seqLengthsMemPtr, MemoryPtr& dstMemPtr) {
+    const VectorDims& srcDims = dataMemPtr->getStaticDims();
+    const auto *srcData = reinterpret_cast<const float *>(dataMemPtr->GetPtr());
+    auto *dstData = reinterpret_cast<float *>(dstMemPtr->GetPtr());
+    auto *seqLengthsData = reinterpret_cast<T *>(seqLengthsMemPtr->GetPtr());
+
+    for (size_t i = 0; i < srcDims[batchAxis]; ++i) {
+        if (static_cast<int32_t>(seqLengthsData[i]) > static_cast<int>(srcDims[seqAxis])) {
+            std::string errorMsg = "Incorrect input 'seq_lengths' values!";
```

Comment:
can be static const, I think. Or maybe even better without variable at all: IE_THROW() << "Incorrect....."

Reply:
done